src/ImageSearchModal.jsx

```diff
@@ -9,244 +9,214 @@ import { Radio } from "@patternfly/react-core/dist/esm/components/Radio";
-                                      type='text'
-                                      placeholder="latest"
-                                      value={this.state.imageTag || ''}
-                                      onChange={(_event, value) => this.onValueChanged('imageTag', value)} />
-                           </FormGroup>
-                       </Form>
-                       <Button variant='primary' isDisabled={this.state.selected == undefined} onClick={this.onDownloadClicked}>
-                           {_("Download")}
-                       </Button>
-                       <Button variant='link' className='btn-cancel' onClick={Dialogs.close}>
-                           {_("Cancel")}
-                       </Button>
-                   </>}
-            >
-                {defaultBody}
-            </Modal>
-        );
-    }
-}
+    };
+
+    const onToggleUser = ev => setIsSystem(ev.currentTarget.value === "system");
+    const onDownloadClicked = () => {
+        const selectedImageName = imageList[selected].Name;
+        if (activeConnection)
+            activeConnection.close();
+        Dialogs.close();
+        downloadImage(selectedImageName, imageTag, isSystem);
+    };
+
+    const defaultBody = (
+        <>
+            <Form isHorizontal>
+                {dialogError && <ErrorNotification errorMessage={dialogError} errorDetail={dialogErrorDetail} />}
+                { userServiceAvailable && systemServiceAvailable &&
+                <FormGroup id="as-user" label={_("Owner")} isInline>
+                    <Radio name="user" value="system" id="system" onChange={onToggleUser} isChecked={isSystem} label={_("system")} />
+                    <Radio name="user" value="user" id="user" onChange={onToggleUser} isChecked={!isSystem} label={user} />
+                </FormGroup>}
+                <Flex spaceItems={{ default: 'inlineFlex', modifier: 'spaceItemsXl' }}>
+                    <FormGroup fieldId="search-image-dialog-name" label={_("Search for")}>
+                        <TextInput id='search-image-dialog-name'
+                                   type='text'
+                                   placeholder={_("Search by name or description")}
+                                   value={imageIdentifier}
+                                   onKeyPress={onKeyPress}
+                                   onChange={(_event, value) => setImageIdentifier(value)} />
+                    </FormGroup>
+                    <FormGroup fieldId="registry-select" label={_("in")}>
+                        <FormSelect id='registry-select'
+                            value={selectedRegistry}
+                            onChange={(_ev, value) => { setSelectedRegistry(value); onSearchTriggered(value, false) }}>
+                            <FormSelectOption value="" key="all" label={_("All registries")} />
+                            {(registries.search || []).map(r => <FormSelectOption value={r} key={r} label={r} />)}
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-1332-20230704-165425-9756b240-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)